conda/cli/main_env_create.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+"""CLI implementation for `conda-env create`.
+
+Creates new conda environments with the specified packages.
+"""
+import json
+import os
+import sys
+import textwrap
+from argparse import RawDescriptionHelpFormatter, _StoreTrueAction
+
+from conda.base.context import context, determine_target_prefix
+from conda.cli import install as cli_install
+from conda.cli.conda_argparse import (
+    add_output_and_prompt_options,
+    add_parser_default_packages,
+    add_parser_networking,
+    add_parser_prefix,
+    add_parser_solver,
+)
+from conda.core.prefix_data import PrefixData
+from conda.deprecations import deprecated
+from conda.env import (
+    specs,
+)
+from conda.env.installers.base import InvalidInstaller, get_installer
+from conda.gateways.disk.delete import rm_rf
+from conda.misc import touch_nonadmin
+from conda.notices import notices
+
+from .common import get_filename, print_result
+
+description = """
+Create an environment based on an environment definition file.
+
+If using an environment.yml file (the default), you can name the
+environment in the first line of the file with 'name: envname' or
+you can specify the environment name in the CLI command using the
+-n/--name argument. The name specified in the CLI will override
+the name specified in the environment.yml file.
+
+Unless you are in the directory containing the environment definition
+file, use -f to specify the file path of the environment definition
+file you want to use.
+"""
+
+example = """
+examples:
+    conda env create
+    conda env create -n envname
+    conda env create folder/envname
+    conda env create -f /path/to/environment.yml
+    conda env create -f /path/to/requirements.txt -n envname
+    conda env create -f /path/to/requirements.txt -p /home/user/envname
+"""
```

Comment:
Let's move all of these constants inline to `configure_parser`, like we do in the other `main_*` scripts